src/components/frame/v5/pages/DashboardPage/partials/ColonyHeader/ColonyHeader.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { FC } from 'react';
+import { useColonyContext } from '~hooks';
+import { formatText } from '~utils/intl';
+import { MAX_TEXT_LENGTH } from './consts';
+import { getCurrentToken } from '~common/ColonyTotalFunds/SelectedToken/helpers';
+import Icon from '~shared/Icon';
+import Tooltip from '~shared/Extensions/Tooltip';
+import Links from '../Links';
+
+const displayName = 'v5.pages.DashboardPage.partials.ColonyHeader';
+
+const ColonyHeader: FC = () => {
+  const { colony } = useColonyContext();
+  const { tokens, nativeToken } = colony || {};
+  const { tokenAddress: nativeTokenAddress } = nativeToken || {};
+  const currentToken = getCurrentToken(tokens, nativeTokenAddress ?? '');
+  const isNativeTokenUnlocked = colony?.status?.nativeToken?.unlocked;
+
+  if (!colony) {
+    return null;
+  }
+  const { metadata } = colony;
+  const description = metadata?.description || '';
+
+  const truncatedText =
+    description.length > MAX_TEXT_LENGTH
+      ? `${description.slice(0, MAX_TEXT_LENGTH - 3)}...`
+      : description;
+
+  return (
+    <header className="flex flex-col gap-4">
+      <div className="flex items-center gap-2">
+        <h1 className="heading-2 text-gray-900 capitalize truncate max-w-[15.625rem] sm:max-w-full">
+          {metadata?.displayName}
+        </h1>
+        <div className="flex items-center gap-2 text-gray-900 bg-base-bg rounded-lg p-1.5">
+          <span className="text-3">{currentToken?.token.symbol}</span>
+          {!isNativeTokenUnlocked && (
+            <Tooltip
+              tooltipContent={
+                <span>{formatText({ id: 'colony.tooltip.lockedToken' })}</span>
+              }
+            >
+              <Icon name="lock-key" appearance={{ size: 'extraTiny' }} />
+            </Tooltip>
+          )}
+        </div>
+      </div>
+      <div className="flex flex-col md:flex-row gap-4 items-start justify-between">
+        {truncatedText ? (
+          <p className="text-gray-700 max-w-[52.75rem] text-md">
+            {truncatedText}
+          </p>
+        ) : (
+          <p className="text-gray-700">
+            {formatText({ id: 'colony.description' })}
+          </p>
+        )}
+        <Links />
```

Comment:
Please add a more descriptive name for this component and also add a wrapper here with `flex-shrink-0` classname